mPower2/mPower2/View Controllers/StudyBurstViewController.swift

```diff
@@ -37,6 +37,7 @@ import ResearchUI
 
 extension RSDIdentifier {
     static let studyBurstCompletionStep: RSDIdentifier = "studyBurstCompletion"
+    static let studyBurstReminderStep: RSDIdentifier = "reminder"
```

Comment:
FYI: 

Since this step is JSON-backed, you can also use the `RSDViewTheme` to point any step at a custom view controller that is in either a NIB or Storyboard --OR-- you can use a custom `type` field to use the factory to instantiate a custom reminder step.

I had to use the other pattern b/c the Demographics survey is vended by the server and the only way to use a custom view controller with server-backed surveys is to look at the step identifier. It's more brittle but didn't require a refactor.

TODO: Refactor to facilitate using this step from the profile without having to also implement the identifier pattern used to special case the intro step on the survey.

Reply:
@syoung-smallwisdom which class vends a Step VC based on 'type'? Can't find it...

Reply:
What I mean is that you could do something like this (localizing as appropriate, of course):

```
extension RSDStepType {
    
    /// Defaults to a `SBAReminderStep`.
    public static let reminder: RSDStepType = "reminder"
}

open class SBAReminderStep : RSDFormUIStepObject {
    
    public init(identifier: String) {
        super.init(identifier: identifier, inputFields: SBAReminderStep.defaultInputFields(), type: .reminder)
    }
    
    public required init(identifier: String, type: RSDStepType?) {
        super.init(identifier: identifier, type: type ?? .reminder)
    }
    
    public required init(from decoder: Decoder) throws {
        try super.init(from: decoder)
    }
    
    private static func defaultInputFields() -> [RSDInputField] {
        let reminderTime = RSDInputFieldObject(identifier: "reminderTime",
                                               dataType: .base(.date),
                                               uiHint: .picker,
                                               prompt: "Set reminder")
        reminderTime.range = RSDDateRangeObject(minuteInterval: 15, defaultTime: "09:00")
        
        let noReminder = RSDInputFieldObject(identifier: "noReminder",
                                             dataType: .base(.boolean),
                                             uiHint: "checkbox",
                                             prompt: "Do not set Reminder")
        
        return [reminderTime, noReminder]
    }
    
    #if !os(watchOS)
    /// Override to return a symptom logging step view controller.
    open func instantiateViewController(with taskPath: RSDTaskPath) -> (UIViewController & RSDStepController)? {
        return ReminderStepViewController(step: self)
    }
    #endif
}

extension SBAReminderStep : RSDStepViewControllerVendor {
}

/// Stub out the factory.
class MP2Factory : SBAFactory {
    
    override func decodeStep(from decoder: Decoder, with type: RSDStepType) throws -> RSDStep? {
        switch type {
        case .reminder:
            return try SBAReminderStep(from: decoder)
        default:
            return try super.decodeStep(from: decoder, with: type)
        }
    }
}
```

And then your JSON looks like this:

```
             {
                "identifier":"reminder",
                "type":"reminder",
                "title":"Let’s set a Reminder for your Study Bursts",
                "detail":"To make it easier for you to remember to do the study burst, you can do the activities during something you normally do every day, such as taking your medications.",
                 "image":{
                     "type": "fetchable",
                     "imageName":"remindersIcon",
                     "placementType":"iconBefore"
                 }
             }
```

Reply:
OR in a simpler case, do this:

```
extension RSDStepType {
    
    /// Defaults to a `ReminderStep`.
    public static let reminder: RSDStepType = "reminder"
}

class ReminderStep : RSDFormUIStepObject {
    
    #if !os(watchOS)
    /// Override to return a symptom logging step view controller.
    func instantiateViewController(with taskPath: RSDTaskPath) -> (UIViewController & RSDStepController)? {
        return ReminderStepViewController(step: self)
    }
    #endif
}

extension SBAReminderStep : RSDStepViewControllerVendor {
}

/// Stub out the factory.
class MP2Factory : SBAFactory {
    
    override func decodeStep(from decoder: Decoder, with type: RSDStepType) throws -> RSDStep? {
        switch type {
        case .reminder:
            return try SBAReminderStep(from: decoder)
        default:
            return try super.decodeStep(from: decoder, with: type)
        }
    }
}
```

Reply:
Since you aren't using a storyboard, you can't use a viewTheme, I don't think. Swift doesn't really support the class from string thing that obj-c supports.

Reply:
Thanks. Yeah, was already doing this following example of 'SBATrackedItemsLoggingStepObject'. Wasn't sure that was the right approach